sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultResourceSpec.java

```diff
@@ -44,14 +31,20 @@ public DefaultResourceSpec(
         this.role = role;
         this.preReservedRole = preReservedRole == null ? Constants.ANY_ROLE : preReservedRole;
         this.principal = principal;
+
+        ValidationUtils.nonEmpty(this, "name", name);
+        ValidationUtils.nonNull(this, "value", value);
+        ValidationUtils.nonEmpty(this, "role", role);
+        ValidationUtils.nonEmpty(this, "principal", principal);
+
+        if (value.hasScalar() && value.getScalar().getValue() <= 0) {
+            throw new IllegalArgumentException(
+                    String.format("Scalar resource value must be greater than zero: %s", this));
```

Comment:
> I figured it could be left ouf of ValidationUtils since it's somewhat specific to resource validation
Yes, by the end of the review, I had come to a similar conclusion.